Dockerfile.horizon

```diff
@@ -0,0 +1,10 @@
+FROM golang:1.19 AS builder
+
+ARG REF
+WORKDIR /go/src/github.com/stellar/go
+RUN git clone https://github.com/stellar/go /go/src/github.com/stellar/go
+RUN git fetch origin ${REF}
+RUN git checkout ${REF}
+ENV CGO_ENABLED=0
+ENV GOFLAGS="-ldflags=-X=github.com/stellar/go/support/app.version=${REF}-(built-from-source)"
+RUN go install github.com/stellar/go/services/horizon
```

Comment:
FYI @ire-and-curses I'm not going to use the Horizon Dockerfile so that the quickstart image has a little more control. I might actually switch all the images to be based locally to this repo, but just doing it with Horizon for now.